elastic/kibana
# [Custom threshold][Alert details page] Deprecate feature flag for Custom threshold rule alert details page

Closes #153867

## Summary

This PR deprecates the following feature flag for the Custom threshold rule alert details page.
```
xpack.observability.unsafe.alertDetails.observability.enabled
```

![image](https://github.com/elastic/kibana/assets/12370520/b0b7bfa4-5f53-48bc-a44c-2f999a3c53f6)


## 🧪 How to test
- Remove `xpack.observability.unsafe.alertDetails.observability.enabled` from Kibana config
- Create a custom threshold rule
- Check the alert details page
    - From alert table's flyout
    - From action variables
- Add `xpack.observability.unsafe.alertDetails.observability.enabled` to the Kibana config
- You should see the following warning
  ```
  [WARN ][config.deprecation] You no longer need to configure "xpack.observability.unsafe.alertDetails.observability.enabled".
  ```

## Release note

Adds a dedicated alert details page for the Custom threshold rule. When a threshold is breached, the alert details page helps users quickly triage the alert, showing details and context around that alert such as when the alert started, its current status, and other details. When a log rate-based threshold is breached, the alert details page also includes an automatic log rate analysis depicting shared characteristics among log messages contributing to this change and in-context Observability AI Assistant to provide further insight and remedy suggestions.
